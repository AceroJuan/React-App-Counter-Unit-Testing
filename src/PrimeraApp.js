import React from "react";
import PropTypes from "prop-types";

const PrimeraApp = ({ saludo, subtitle }) => {
  /* const saludo = {
    nombre: "Juan",
    edad: 36,
  }; */

  return (
    <>
      <h1>{saludo}</h1>
      {/* <h1>{JSON.stringify(saludo, null, 3)}</h1> */}
      <p>{subtitle}</p>
    </>
  );
};
PrimeraApp.propTypes = {
  saludo: PropTypes.string.isRequired,
};
PrimeraApp.defaultProps = {
  subtitle: "Soy un subtitulo viteh",
};

export default PrimeraApp;
