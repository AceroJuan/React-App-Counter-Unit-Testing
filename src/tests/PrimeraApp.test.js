// import { render } from "@testing-library/react";
import React from "react";
import PrimeraApp from "../PrimeraApp";
import { shallow } from "enzyme";

describe("pruebas para PrimeraApp.js", () => {
  //   test("debe mostrar el mensaje 'Hola, soy Goku!'", () => {
  //     const saludo = "Hola, soy Goku!";
  //     const { getByText } = render(<PrimeraApp saludo={saludo} />);
  //     expect(getByText(saludo + "!!!")).toBeInTheDocument();
  //   });
  test("debe de mostrar '<PriemeraApp />' correctamente", () => {
    const saludo = "Hola, soy Goku!";
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("debe mostrar el subtitulo enviado por props", () => {
    const saludo = "Hola, soy Goku!";
    const subTitulo = "hola soy un subtitulo nuevo";
    const wrapper = shallow(
      <PrimeraApp saludo={saludo} subtitle={subTitulo} />
    );
    const textParragraph = wrapper.find("p").text();
    expect(textParragraph).toBe(subTitulo);
  });
});
