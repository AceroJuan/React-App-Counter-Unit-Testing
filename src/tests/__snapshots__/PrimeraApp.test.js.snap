// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas para PrimeraApp.js debe de mostrar '<PriemeraApp />' correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy Goku!
  </h1>
  <p>
    Soy un subtitulo viteh
  </p>
</Fragment>
`;
